page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/StashParser.java

```diff
@@ -0,0 +1,142 @@
+                        argOffset = stepDataMatcher.start();
+                        offset = stashValue.length() + stashKeyFullPath.length();
+                    }
+
+                    replacedValue = replacedValue.replace(stepDataMatcher.start(), stepDataMatcher.end(), stashValue);
+                    stepDataMatcher = stepDataPattern.matcher(replacedValue);
+                } else {
+                    argOffset = argument.getOffset() - offset;
+                }
+            }
+            replacedArguments.add(new Argument(argOffset, argVal));
+        }
+        FieldUtils.writeField(FieldUtils.readField(testStep, "definitionMatch", true),
+                "arguments", replacedArguments, true);
+    }
+
+    private static void replaceStepText(PickleStepCustom currentStep) throws IllegalAccessException {
+        currentStep.setText(StashParser.replaceDataPlaceholders(currentStep, currentStep.getText()));
+    }
+
+    private static void replacePickleArguments(PickleStepCustom currentStep) throws IllegalAccessException {
+        for (gherkin.pickles.Argument argument : currentStep.getArgument()) {
+            if (argument.getClass().equals(PickleTable.class)) {
+                replaceDataTable(currentStep, (PickleTable) argument);
+            } else {
+                if (argument.getClass().equals(PickleString.class)) {
+                    String content = replaceDataPlaceholders(currentStep, ((PickleString) argument).getContent());
+                    FieldUtils.writeField(argument, "content", content, true);
+                }
+            }
+        }
+    }
+
+    private static void replaceDataTable(PickleStepCustom currentStep, PickleTable argument) throws IllegalAccessException {
+        for (PickleRow pickleRow : argument.getRows()) {
+            for (PickleCell pickleCell : pickleRow.getCells()) {
+                FieldUtils.writeField(pickleCell, "value", replaceDataPlaceholders(currentStep, pickleCell.getValue()), true);
+            }
+        }
+    }
+
+    private static String replaceDataPlaceholders(PickleStepCustom currentStep, String replaceableValue) {
+        Pattern stepDataPattern = Pattern.compile(PATH_PARSE_REGEX);
+        Matcher stepDataMatcher = stepDataPattern.matcher(replaceableValue);
+        StringBuilder replacedValue = new StringBuilder(replaceableValue);
+
+        while (stepDataMatcher.find()) {
+            String stashKey = stepDataMatcher.group(1);
+            String stashValue = getStashValue(currentStep, stashKey);
+            if (stashValue == null) {
+                break;
+            }
+            replacedValue = replacedValue.replace(stepDataMatcher.start(), stepDataMatcher.end(), stashValue);
+            stepDataMatcher = stepDataPattern.matcher(replacedValue);
+        }
+        return replacedValue.toString();
+    }
+
+    private static String getStashValue(PickleStepCustom currentStep, String stashKey) {
+        Object stashValue = Stash.getValue(stashKey);
```

Comment:
String result = stashValue != null && stashValue instanceof String ? (String) stashValue : null;
    String messageTemplate = stashValue == null && result == null ? "Stash value not found by key: " : "The value received by the key must be a string. Key: ";

    saveMessage(currentStep, message + stashKey);
    return result;